proposals/lip-0053.md

```diff
@@ -438,26 +679,20 @@ TBA
 
 ### Computing the Validators Update
 
-When posting a CCU transaction, the validators hash given in the certificate certifies the new set of validators of the sending chain. The CCU must therefore include the difference between the validators currently stored in the chain account and the validator set authenticated by the certificate. The required `activeValidatorsUpdate` can be obtained by applying the function below. We suppose that `currentValidators` and `newValidators` follow the `activeValidators` schema given above.
+When posting a CCU transaction, the validators hash given in the certificate certifies the new set of validators of the sending chain. The CCU must therefore include the difference between the validators currently stored in the chain account and the validator set authenticated by the certificate. The required `activeValidatorsUpdate` can be obtained by applying the function below.
 
 ```python
-getActiveValidatorsDiff(currentValidators, newValidators):
+def getActiveValidatorsDiff(currentValidators: list[ActiveValidator], newValidators: list[ActiveValidator]) -> list[ActiveValidator]:
     activeValidatorsUpdate = []
 
-    for newValidator in newValidators.activeValidators:
-        if there exist currentValidator in currentValidators.activeValdators such that
-        newValidator.blsKey == currentValidator.blsKey
-	and newValidator.bftWeight == currentValidator.bftWeight:
-            continue
-        else:
-            append newValidator to activeValidatorsUpdate
+    for validator in newValidators:
+        if validator not in currentValidators:
```

Comment:
It is not clear what equality means here as two complex objects can have the same values for each property but may still be different as they are different instances. So what we mean here is that we add it unless a validator with the same BLS key and BFT weight already exists.

Reply:
indeed, that is what we mean. I think this just works in python.

Reply:
Could we add a comment to make this clear?

Reply:
added

Reply:
Looks good.